### PR TITLE
Feat(eos_cli_config_gen): Add encapsulation to flow tracking

### LINKED ISSUE #3395: Feat(eos_cli_config_gen): Add encapsulation to flow tracking
### Enhancement summary

Add  encapsulation to flow tracking 

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

```
flow_tracking:
 sampled: 
    encapsulation:
       ipv4: <bool>
       ipv6: <bool>
```

### Describe the solution you would like

Ideally extend the data model to support more flow tracking options.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Add encapsulation to flow tracking

## Related Issue(s)

Fixes #3395 

## Component(s) name

`arista.avd.eos_cli_config_gen`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by chetryan** [comment]
There is no need for any offload features to be added. The title can be updated to just say `Add encapsulation to flow tracking`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_tracking.schema.yml [hunk 1]
```diff
@@ -13,6 +13,13 @@ keys:
         type: dict
         $ref: "eos_cli_config_gen#/$defs/flow_tracking"
         keys:
+          encapsulation:
+            type: dict
+            keys:
+              ipv4:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_tracking.schema.yml]: Let's add `mpls` also as an option

```
border1-dc1(config-flow-tracking-sampled)#show cli command | grep encapsulation
[no|default] encapsulation ( { ENCAPS } )
border1-dc1(config-flow-tracking-sampled)#encapsulation ?
  ipv4  IPv4 flows
  ipv6  IPv6 flows
  mpls  MPLS flows

border1-dc1(config-flow-tracking-sampled)#encapsulation ipv4 ipv6 mpls
border1-dc1(config-flow-tracking-sampled)#exit
border1-dc1(config)#show run | se flow
flow tracking sampled
   encapsulation ipv4 ipv6 mpls
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/flow-tracking.yml [hunk 2]
```diff
@@ -3,6 +3,8 @@
 
 flow_tracking:
   sampled:
+    encapsulation:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/flow-tracking.yml:6]: can we try to exercise multiple options?

**Review by gmuloc — DISMISSED** [review]

**Comment by gmuloc** [comment]
let's add something in documentation as well
https://github.com/aristanetworks/ansible-avd/blob/dcbc841b607e919414a6160e5ec156ceb27ed265/ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-tracking.j2#L15

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/flow-tracking.yml (hunk 2 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/flow-tracking.yml:6]: added mpls as well

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_tracking.schema.yml (hunk 1 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/flow_tracking.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-tracking.j2 [hunk 3]
```diff
@@ -12,19 +12,31 @@
 
 #### Flow Tracking Sampled
 
-| Sample Size | Minimum Sample Size | Hardware Offload for IPv4 | Hardware Offload for IPv6 |
-| ----------- | ------------------- | ------------------------- | ------------------------- |
+| Sample Size | Minimum Sample Size | Hardware Offload for IPv4 | Hardware Offload for IPv6 | Encapsulation for IPv4 | Encapsulation for IPv6 | Encapsulation for MPLS |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-tracking.j2]: I think a single field can work here, and then just add them as comma separated strings like `ipv4, mpls` and default to `-`.
```suggestion
| Sample Size | Minimum Sample Size | Hardware Offload for IPv4 | Hardware Offload for IPv6 | Encapsulations |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-tracking.j2 [hunk 4]
```diff
@@ -12,19 +12,31 @@
...
+| ----------- | ------------------- | ------------------------- | ------------------------- | ---------------------- | ---------------------- | ---------------------- |
 {%         set sample_size = flow_tracking.sampled.sample | arista.avd.default(flow_trackings[0].sample, "default") %}
 {%         set min_sample_size = flow_tracking.sampled.hardware_offload.threshold_minimum | arista.avd.default("default") %}
 {%         set hardware_offload_ipv4 = 'disabled' %}
 {%         set hardware_offload_ipv6 = 'disabled' %}
+{%         set encapsulation_ipv4 = 'disabled' %}
+{%         set encapsulation_ipv6 = 'disabled' %}
+{%         set encapsulation_mpls = 'disabled' %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-tracking.j2]: If you go with my suggestion above with a single column, you can do
```suggestion
{%         set encapsulations = ([flow_tracking.sampled.encapsulation.ipv4 | arista.avd.default, flow_tracking.sampled.encapsulation.ipv6 | arista.avd.default, flow_tracking.sampled.encapsulation.mpls | arista.avd.default] | select("arista.avd.defined") or ["-"]) | join(", ") %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-tracking.j2 (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-tracking.j2]: Or just use the simpler logic similar to your CLI template.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-tracking.j2 [hunk 5]
```diff
@@ -12,19 +12,30 @@
...
+{%         if flow_tracking.sampled.encapsulation.ipv4 is arista.avd.defined(true) %}
+{%             do encapsulations_list.append('ipv4') %}
+{%         endif %}
+{%         if flow_tracking.sampled.encapsulation.ipv6 is arista.avd.defined(true) %}
+{%             do encapsulations_list.append('ipv6') %}
+{%         endif %}
+{%         if flow_tracking.sampled.encapsulation.mpls is arista.avd.defined(true) %}
+{%             do encapsulations_list.append('mpls') %}
+{%         endif %}
+{%         set encapsulations = (encapsulations_list | select("arista.avd.defined") or ["-"]) | join(", ") %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/flow-tracking.j2]:
```suggestion
{%         set encapsulations = (encapsulations_list or ["-"]) | join(", ") %}
```

**Comment by Shivani-gslab** [comment]
Thanks @ClausHolbechArista  for fixing this.

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md [hunk 6]
```diff
@@ -91,6 +91,7 @@ interface Management1
 ```eos
 !
 flow tracking sampled
+   encapsulation ipv4 mpls
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md]: This is being rejected on my test device (apologies for not testing this further before):

```
ceos1(config-flow-tracking-sampled)#encapsulation mpls
% Both ipv4 and ipv6 encapsulations must be set
ceos1(config-flow-tracking-sampled)#encapsulation ipv6 mpls
% Both ipv4 and ipv6 encapsulations must be set
ceos1(config-flow-tracking-sampled)#encapsulation ipv4 mpls
% Both ipv4 and ipv6 encapsulations must be set
```

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md]: Looks like some combinations are not allowed so we may need to change the model

**Review by gmuloc — DISMISSED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md (hunk 6 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md]: so we can change the key to ipv4_ipv6_mpls :)

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md]: and `ipv4_ipv6` probably need to recheck CLI

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md]:
```
s1-spine1(config-flow-tracking-sampled)#encapsulation ipv6
% Both ipv4 and ipv6 encapsulations must be set
s1-spine1(config-flow-tracking-sampled)#encapsulation ipv4
% Both ipv4 and ipv6 encapsulations must be set
s1-spine1(config-flow-tracking-sampled)#encapsulation ipv4 ipv6
s1-spine1(config-flow-tracking-sampled)#
```
So `ipv4 ipv6` and `ipv4 ipv6 mpls` combinations are allowed.
We can have keys - `ipv4_ipv6` and `mpls` ?

Comment (carlbuchmann) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/flow-tracking.md]: yes `ipv4_ipv6` and `mpls` would be correct here!

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**